### PR TITLE
{AKS} Update tests for ASM egress gateway to allow for BadRequest response

### PR DESCRIPTION
**Pull request by deveshdama**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
`az aks mesh enable-egress-gateway --resource-group MyResourceGroup --name MyManagedCluster`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -11588,57 +11589,69 @@ def test_aks_azure_service_mesh_with_egress_gateway(
             "aks mesh enable-egress-gateway --resource-group={resource_group} --name={name} "
             "--egress-gateway-nodeselector istio=egress"
```
Comment (nshankar13) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:11590]: @biefy wondering if we should also remove the --nodeselector param from these tests, just to be safe. Or should we do that in a follow-up?

**Review by FumingZhang — APPROVED** [review]
lgtm

Feature owner queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=88544103&view=results), test passed!

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py (hunk 1 above)
Comment (nshankar13) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:11590]: In retrospect, let's finalize the design first and then confirm what to do about nodeSelector first.

**Pull request merged**